### DISCUSSION TITLE
How to send a document to printer

### DISCUSSION THREAD

**Discussion by cbrunin** [AvaloniaUI/Avalonia#7763] [Q&A]
Hi,
We are using Avalonia to build a multiplatform app. One of our needs is to print some documents (PDF -> printer).
Is it supported by Avalonia? Any advise?
Thanks!
Chris

**Comment by timunie** [comment]
I think it's not Avalonia related but .NET related. Maybe this thread on Stackoverflow helps you

https://stackoverflow.com/q/273675/13051114

Happy coding
Tim

**Comment by cbrunin** [comment]
Hi Tim,

Thanks for your feedback,
PrinterSettings is a member of System.Drawing which is not multiplatform. That's the problem we are facing.

Chris

**Reply by robloo** [reply]
> Is it supported by Avalonia?

Avalonia does not support printing out of the box. However, it is possible to do natively for each platform using Xamarin. You'll have to find examples online: which formats to use and how to pass them to the platform-specific native APIs.

**Reply by robloo** [reply]
Here is an example. This is a quick search -- haven't tried it myself.

https://www.pujolsluis.com/printing-pdfs-and-images-xamarin-forms/

**Comment by FWDAzhilay** [comment]
Hello! Have you found a way to print from the Avalonia box?

I would be grateful if you share your experience.

**Reply by thevortexcloud** [reply]
Pretty much anything involving printers is complicated. On Windows I don't think there is any simple way to send something directly to a printer? Most people don't bother and just use an existing library or shell out to another program (EG Adobe acrobat). On Mac/Linux you can technically use CUPS which is _significantly_ simpler.

**Reply by AlanPBourke** [reply]
https://github.com/jesusvlo/RawPrint.NetStd
